usersimcrs/items/item.py

```diff
@@ -55,5 +73,12 @@ def set_property(self, key: str, value: Any) -> None:
         Args:
             key: Property name.
             value: Property value.
+
+        Raises:
+            ValueError: if the property is not part of the domain knowledge.
         """
+        if self._domain and key not in self._slot_names:
+            raise ValueError(
+                f"The property {key} is not part of the Domain classes."
```

Comment:
Nit: "... is not part of the slots specified by the domain."

Reply:
Done.

Reply:
Done.